lib/logstash/filters/dns.rb

```diff
@@ -219,8 +226,13 @@ def resolve(event)
         end
       rescue Resolv::ResolvTimeout
         @failed_cache[raw] = true if @failed_cache
-        @logger.warn("DNS: timeout on resolving the hostname.",
-                      :field => field, :value => raw)
+        log_message = "DNS: timeout on resolving the hostname."
+        if @warn_on_timeout
+          @logger.warn(log_message, :field => field, :value => raw)
+        else
+          @logger.info(log_message, :field => field, :value => raw)
+        end
```

Comment:
it seems this change tries to address t[his issue](https://github.com/logstash-plugins/logstash-filter-dns/issues/47).

The requested feature is asking for a wait to _**disable**_ logging. It is NOT asking for a way to change this logging to from WARN to INFO. Can you please clarify why is this change relevant to the attached issue?

Reply:
another option is to make this DEBUG as the original PR.

Reply:
Long history and it is true that now the logic is adding _extra noise_ :( and doesn't solve the user need.
Stepping back and analyzing, I agree with the _debug_ approach as we added the tag. Users will have two granular options: receiving with tag and debug logs.
Please take a look at the final approach and provide your comments.
Thanks so much for catching this up!